internal/api/create_or_update_resource_v1alpha1.go

```diff
@@ -42,6 +42,14 @@ func (s *server) createOrUpdateResource(
 	ctx context.Context,
 	obj *unstructured.Unstructured,
 ) (*svcv1alpha1.CreateOrUpdateResourceResult, error) {
+	if obj.GroupVersionKind() == secretGVK && !s.cfg.SecretManagementEnabled {
```

Comment:
not very familiar with this comparison but just checking, does it compare deep equal?

Reply:
`==` comparison in Go isn't deep. It checks equality field-by-field. This type is composed of only three `string` fields, so although this comparison isn't deep, it's as deep as it needs to be.